apollo-router/src/services/subgraph_service.rs

```diff
@@ -291,4 +430,46 @@ mod tests {
             "HTTP fetch failed from 'test': subgraph didn't return JSON (expected content-type: application/json or content-type: application/graphql+json; found content-type: \"text/html\")"
         );
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_compressed_request_response_body() {
+        let socket_addr = SocketAddr::from_str("127.0.0.1:2727").unwrap();
```

Comment:
Nit/followup: We might want to bind on port 0 and let the machine decide where to spawn the server (not important at all because I don't expect two tests to run in parallel tbh)